### ISSUE TITLE
Incomplete covariance matrices 

### ISSUE THREAD

**Issue by esrah-ap-niht** [borchero/pycave#10]
Hey Oliver, thank you very much for making PyCave, I think it's going to make my life much easier compared to Sklean. 

I think the covariance matrices are not being calculated or stored correctly. For example, I tested a model with 20 components/clusters and 8 features per data point. The covariance matrix in `gmm.gaussian.covars` output as a 1X20 array when it should be 8X8X20. The component weights and mixture means appear to be correct. 

![image](https://user-images.githubusercontent.com/86078445/122461415-5349ac00-cf81-11eb-980b-81e4b21e535a.png)

**Comment by groadabike** [comment]
Hi @esrah-ap-niht ,
Are you using diagonal covariance?
If so, the shape is correct.
https://github.com/borchero/pycave/blob/691179be4340ec3f9757faf3c73dfbab6200490e/pycave/bayes/gmm/model.py#L57

**Comment by esrah-ap-niht** [comment]
@groadabike 
No, I had `gmm = GMM(num_components=20, num_features=test2.shape[1], covariance='spherical')` specified. However, you are absolutely correct that the specification explains the shape of the matrix. I feel stupid I didn't catch that myself. 

Thinking about this, the thread title should actually change to a feature request for a full covariance matrix option. Or if that's already implemented, knowing what text to use in the specification. I tried using "full" in the same style as Sklearn and received an error message. 

![image](https://user-images.githubusercontent.com/86078445/122466403-43cd6180-cf87-11eb-8e6d-19bb28309fd6.png)

**Comment by borchero** [comment]
Hey @esrah-ap-niht. Unfortunately, full covariance matrices are currently not implemented. I'm in the process of refactoring the library though to make the interface even closer to sklearn's. I'll make sure to add it then :)

**Comment by esrah-ap-niht** [comment]
Thank you @borchero, I'll look forward to it. I'm using your code for research and would like to cite your work. Do you have a reference you'd like me to use? Otherwise I can use a generic reference along the lines of: 

![image](https://user-images.githubusercontent.com/86078445/122566178-41194d80-d015-11eb-8d81-ca0b21d1b6b0.png)

**Comment by borchero** [comment]
Happy to see it being used in your research, @esrah-ap-niht ! :D unfortunately, there is no public work using the library, so the generic reference is definitely fine :) Thanks for asking!
